modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/shelf/PlayerShelf.kt

```diff
@@ -0,0 +1,530 @@
+                    onClick = onDownloadClick,
+                )
+                ShelfItem.Share -> ShareButton(
+                    iconColors = iconColors,
+                    onClick = onShareClick,
+                )
+                ShelfItem.Podcast -> PodcastButton(
+                    iconColors = iconColors,
+                    onClick = onShowPodcast,
+                )
+                ShelfItem.Cast -> CastButton(
+                    iconColors = iconColors,
+                    onClick = onCastClick,
+                )
+                ShelfItem.Played -> PlayedButton(
+                    iconColors = iconColors,
+                    onClick = onPlayedClick,
+                )
+                ShelfItem.Bookmark -> BookmarkButton(
+                    iconColors = iconColors,
+                    onClick = onAddBookmarkClick,
+                )
+                ShelfItem.Archive -> ArchiveButton(
+                    isUserEpisode = playerShelfData.isUserEpisode,
+                    iconColors = iconColors,
+                    onClick = onArchiveClick,
+                )
+                ShelfItem.Report -> ReportButton(
+                    iconColors = iconColors,
+                    onClick = onReportClick,
+                )
+            }
+        }
+        MoreButton(
+            iconColors = iconColors,
+            onClick = onMoreClick,
+        )
+    }
+}
+
+@Composable
+fun EffectsButton(
+    isEffectsOn: Boolean,
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    val effectsTint =
+        if (isEffectsOn) iconColors.highlightColor else iconColors.normalColor
+    val effectsResource =
+        if (isEffectsOn) R.drawable.ic_effects_on_32 else R.drawable.ic_effects_off_32
+    IconButton(onClick = onClick) {
+        Icon(
+            painterResource(id = effectsResource),
+            contentDescription = stringResource(LR.string.player_effects),
+            tint = effectsTint,
+        )
+    }
+}
+
+@Composable
+fun SleepButton(
+    isSleepRunning: Boolean = false,
+    iconColors: PlayerShelfIconColors,
+    onClick: () -> Unit,
+) {
+    val sleepTint =
+        if (isSleepRunning) iconColors.highlightColor else iconColors.normalColor
+    val alpha = if (isSleepRunning) 1F else AndroidColor.alpha(sleepTint.toArgb()) / 255F * 2F
+    val lottieComposition by rememberLottieComposition(LottieCompositionSpec.RawRes(R.raw.sleep_button))
+    val progress by animateLottieCompositionAsState(
+        lottieComposition,
+        iterations = LottieConstants.IterateForever,
+        isPlaying = isSleepRunning,
+    )
+    val dynamicProperties = rememberLottieDynamicProperties(
+        rememberLottieDynamicProperty(
+            property = LottieProperty.COLOR_FILTER,
+            value = SimpleColorFilter(sleepTint.toArgb()),
+            keyPath = arrayOf("**"),
+        ),
+    )
+    IconButton(
+        onClick = onClick,
+        modifier = Modifier.alpha(alpha),
```

Comment:
<img  width="300" src="https://github.com/user-attachments/assets/bb83f616-5dca-4fcb-b5a3-e3c2026667e6" /> 

This is missing a TalkBack label.